### PR TITLE
gh-116116: Backport PR #42 to fix building with clang-cl on windows-i686

### PR DESCRIPTION
**Pull request by georgthegreat**
Backport [PR #42](https://github.com/blake2/libb2/pull/42) to work around the following compilation error:
```
In file included from $(SOURCE_ROOT)/python3/src/Modules/_blake2/blake2b_impl.c:30:
$(SOURCE_ROOT)/python3/src/Modules/_blake2/impl/blake2b.c(31,23): error: conflicting types for '_mm_set_epi64x'
static inline __m128i _mm_set_epi64x( const uint64_t u1, const uint64_t u0 )
                      ^
$(TOOL_ROOT)/clang/14.0.6/include/emmintrin.h(3613,1): note: previous definition is here
_mm_set_epi64x(long long __q1, long long __q0)
^
1 error generated.
```

* Issue: gh-116116

### CODE REVIEW

**Comment by zooba** [comment]
Seems fine to me. The generated files are the SBOM containing the hashes of these files.

@SethMichaelLarson What's the way we should handle this kind of in-repo patching? Is regenerating the SBOM enough? Or do we need to track the before/after state as well?

**Comment by sethmlarson** [comment]
@zooba I think it's fine that we make minor in-source changes, we run the risk of losing them when the dependency is updated so we should try to upstream the changes too. In a perfect world we'd be marking our SBOM with precise patches/descendants relationships, but I'm not sure how important that is for our primary use-case which is vulnerability management/discovery.

**Comment by zooba** [comment]
Okay, so this one is a backport from upstream so we won't lose it. We just need the SBOM to be regenerated to be able to merge the PR. I believe there's an open PR adding that functionality to the Windows build, so we'll probably have to wait for that to be merged before this can be updated.

**Comment by georgthegreat** [comment]
@zooba, I have updated sbom files as requested

**Comment by georgthegreat** [comment]
@zooba, let's merge it.

**Review by gpshead — APPROVED** [review]

**Comment by zooba** [comment]
@georgthegreat Is this needed for 3.11? Could you prepare a PR for that branch? I suspect it's the SBOM conflicting, but maybe it's not needed for an older version of blake2.

**Comment by zooba** [comment]
> Do you consider this bug critical?

Nope :) But it wasn't critical to me in main either, so that's not a great bar for whether to backport. Happy to defer to your needs, and if someone else needs it for 3.11 then they can prepare a backport.

**Pull request merged**
